vitality/database.py

```diff
@@ -481,6 +481,54 @@ def set_workout_difficulty(self, id: str, difficulty: str):
                 }
             })
 
+    def set_workout_total_time(self, creator_id: str, name: str, total_time: str):
+        """Updates a workout's total time given a workout id."""
+        self.mongo.workout.update_one(
+            {"creator_id": ObjectId(creator_id),
+             "name": name
+             },
+            {
+                "$set": {
+                    "total_time": total_time
+                }
+            })
+
+    def set_workout_reps(self, creator_id: str, name: str, reps: str):
+        """Updates a workout's reps given a workout id."""
+        self.mongo.workout.update_one(
+            {"creator_id": ObjectId(creator_id),
+             "name": name
+             },
+            {
+                "$set": {
+                    "reps": reps
+                }
+            })
+
+    def set_workout_miles(self, creator_id: str, name: str, miles: str):
+        """Updates a workout's miles given a workout id."""
+        self.mongo.workout.update_one(
+            {"creator_id": ObjectId(creator_id),
+             "name": name
+             },
+            {
+                "$set": {
+                    "miles": miles
+                }
+            })
+
+    def set_workout_category(self, creator_id: str, name: str, category: str):
+        """Updates a workout's category given a workout id."""
+        self.mongo.workout.update_one(
+            {"creator_id": ObjectId(creator_id),
+             "name": name
+             },
+            {
+                "$set": {
+                    "category": category
+                }
+            })
+
```

Comment:
Need unit tests under test_database.py for these functions.

Reply:
Needs to be functions `test_set_workout_total_time, set_workout_reps, set_workout_miles, set_workout_category`.